app/data_client.go

```diff
@@ -430,12 +517,12 @@ func (d *DataClient) TabularDataByFilter(
 	}
 	// TabularData contains tabular data and associated metadata
 	dataArray := []TabularData{}
-	var metadata *pb.CaptureMetadata
 	for _, data := range resp.Data {
-		if len(resp.Metadata) > 0 && int(data.MetadataIndex) < len(resp.Metadata) {
+		var metadata *pb.CaptureMetadata
+		switch {
+		case len(resp.Metadata) > 0 && int(data.MetadataIndex) < len(resp.Metadata):
```

Comment:
(minor) because `MetadataIndex` is an unsigned int, the first check (`len(resp.Metadata) > 0`) is superfluous. If `len(resp.Metadata) == 0` then by necessity `int(data.MetadataIndex) >= len(resp.Metadata)`.